### PR TITLE
[ECS] Add ``data/opentelekomcloud_compute_instance_v2``

### PR DESCRIPTION
**Pull request by vladimirvshivkov**
## PR Checklist

* [x] Refers to: #1771 
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccComputeV2InstanceDataSource_basic
--- PASS: TestAccComputeV2InstanceDataSource_basic (133.60s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

**Review by anton-sidelnikov — DISMISSED** [review]

File: opentelekomcloud/acceptance/ecs/data_source_opentelekomcloud_compute_instance_v2_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/acceptance/common"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/acceptance/env"
+)
+
+func TestAccComputeV2InstanceDataSource_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			common.TestAccPreCheck(t)
+			common.TestAccPreCheckAdminOnly(t)
```
Comment (lego963) [inline opentelekomcloud/acceptance/ecs/data_source_opentelekomcloud_compute_instance_v2_test.go]: is it required?

File: docs/data-sources/ecs_instance_v1.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+---
+
+# opentelekomcloud_ecs_instance_v1
+
+Get information on an ECS instance.
+
+## Example Usage
+
+```hcl
+data "opentelekomcloud_ecs_instance_v1" "instance" {
```
Comment (lego963) [inline docs/data-sources/ecs_instance_v1.md]: is it ecs_instance or compute_instance?

File: opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+		},
+	}
+}
+
+func dataSourceComputeInstanceV2Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	log.Print("[DEBUG] Creating compute client")
+	client, err := config.ComputeV2Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("Error creating compute client: %s", err)
```
Comment (lego963) [inline opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go]: use errCreateV2Client

File: opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+			},
+			"power_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
```
Comment (lego963) [inline opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go]: use common.TagsSchema()

File: opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+	d.SetId(server.ID)
+
+	mErr := multierror.Append(
+		d.Set("name", server.Name),
+		d.Set("image_id", server.Image["ID"]),
+	)
+	if err := mErr.ErrorOrNil(); err != nil {
+		return diag.FromErr(err)
+	}
```
Comment (lego963) [inline opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go]: duplicate code, you can drop it

File: opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+	err = servers.Get(client, d.Id()).ExtractInto(&serverWithAZ)
+	if err != nil {
+		return diag.FromErr(common.CheckDeleted(d, err, "server"))
+	}
+
+	// Set the current power_state
+	currentStatus := strings.ToLower(server.Status)
+	switch currentStatus {
+	case "active", "shutoff", "error", "migrating", "shelved_offloaded", "shelved":
+		_ = d.Set("power_state", currentStatus)
```
Comment (lego963) [inline opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go]: do not ingnore errors

File: opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+	if err := mErr.ErrorOrNil(); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Populate tags
+	instanceTags, err := tags.List(client, server.ID).Extract()
+	if err != nil {
+		log.Printf("[DEBUG] Unable to get tags for openstack_compute_instance_v2: %s", err)
+	} else {
+		_ = d.Set("tags", instanceTags)
```
Comment (lego963) [inline opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go]: same with errors

File: opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+		d.Set("flavor_name", flavor.Name),
+		d.Set("availability_zone", serverWithAZ.AvailabilityZone),
+		d.Set("region", config.GetRegion(d)),
+	)
+	if err := mErr.ErrorOrNil(); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Populate tags
+	instanceTags, err := tags.List(client, server.ID).Extract()
```
Comment (lego963) [inline opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go]: is it working?
and 
// List is a method of getting the tags of all service

**Review by lego963 — CHANGES_REQUESTED** [review]

File: opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go (hunk 8 above)
Comment (vladimirvshivkov) [reply opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_instance_v2.go]: good question)

**Review by lego963 — APPROVED** [review]

**Pull request merged**